tx.go

```diff
@@ -104,7 +104,7 @@ func newTx(db *DB, writable bool) (tx *Tx, err error) {
 	tx = &Tx{
 		db:                     db,
 		writable:               writable,
-		pendingWrites:          []*Entry{},
+		pendingWrites:          make(map[string][]*Entry),
```

Comment:
Just creating an object of your special Type is enough.This will make your code more elegant.